src/NLog.Targets.Syslog/ByteArray.cs

```diff
@@ -37,6 +41,21 @@ public ByteArray(long initialCapacity)
             }
         }
 
+        public void Append(string buffer, Encoding encoding)
+        {
+            if (string.IsNullOrEmpty(buffer))
+                return;
+
+            var byteCount = encoding.GetByteCount(buffer);
+            memoryStream.SetLength(memoryStream.Length + byteCount);
+            for (int i = 0; i < buffer.Length; i += encodingBuffer.Length)
+            {
+                int remainingCount = Math.Min(buffer.Length - i, encodingBuffer.Length);
+                buffer.CopyTo(i, encodingBuffer, 0, remainingCount);
+                memoryStream.Position += encoding.GetBytes(encodingBuffer, 0, remainingCount, memoryStream.GetBuffer(), (int)memoryStream.Position);
+            }
+        }
+
```

Comment:
This idea is good, despite reducing readability where this method is called.
The implementation with a fixed buffer size and for doesn't allow, in my opinion, to have great performance advantages, but luckily this can be done operating directly on strings:
```csharp
public void Append(string s, Encoding encoding)
{
    if (string.IsNullOrEmpty(s))
        return;

    memoryStream.Position += encoding.GetBytes(s, 0, s.Length, memoryStream.GetBuffer(), (int)memoryStream.Position);    
```
When the memoryStream capacity is int.MaxValue and we try writing more than the room left an exception will be thrown: we take this risk without adding protective code since it will be difficult, if not impossible, for a client application, to log a string that is int.MaxValue chars long.

Reply:
"Luckily" this is a trap. One should never give the string directly to Encoding-object. See the ToCharArray()-call here: https://referencesource.microsoft.com/#mscorlib/system/text/encoding.cs,1075 My version works better.

Reply:
I saw the ToCharArray before writing your previous comment, but I was trying to find a more readable way to do the same thing: yes you do not allocate, but you copy in a char array to then getbytes...

Probably it would be better to do something like this:
```csharp
public void Append(string s, Encoding encoding)
{
    if (string.IsNullOrEmpty(s))
        return;

    var length = s.Length;
    var byteCount = encoding.GetByteCount(s):
    unsafe
    {
        fixed (char* chars = s)
        fixed (byte* buffer = memoryStream.GetBuffer())
        {
            memoryStream.Position += encoding.GetBytes(chars, length, buffer, byteCount);
    }
}
```

Reply:
I'm not a big fan of unsafe code, but if you like that then please go ahead.